cmd/sriov-network-config-daemon/start.go

```diff
@@ -294,25 +296,47 @@ func runStartCmd(cmd *cobra.Command, args []string) error {
 	log.Log.Info("Enabled featureGates", "featureGates", featureGates.String())
 
 	setupLog.V(0).Info("Starting SriovNetworkConfigDaemon")
-	err = daemon.New(
-		kClient,
-		snclient,
-		kubeclient,
-		hostHelpers,
-		platformHelper,
-		exitCh,
-		stopCh,
-		syncCh,
-		refreshCh,
-		eventRecorder,
-		featureGates,
-		startOpts.disabledPlugins,
-	).Run(stopCh, exitCh)
-	if err != nil {
-		setupLog.Error(err, "failed to run daemon")
+
+	// create a signal channel to catch interrupts
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt)
+	signal.Notify(sigc, syscall.SIGTERM)
+
+	errChan := make(chan error)
+	defer close(errChan)
```

Comment:
I used the method used here to close channels. But I believe there is a race in general, on exit (once main goes away, or if os.Exit is called) the goroutines are killed off immediately. So the goroutines listening on this channel will never get the chance to run their intended actions.